api/utils/filters.py

```diff
@@ -32,29 +32,33 @@ def get_time_filtered_temperatura(request):
 
 
 def get_time_filtered_pauta(request):
+    '''
+    Filtra as pautas que ocorreram depois ou no dia da data_referencia. Se a
+    data_referencia é uma sexta-feira, retornará todas as pautas com um
+    período de 6 dias após, se não,retorna as pautas da mesma semana da data_referencia.
+    '''
     data_referencia = request.query_params.get('data_referencia')
-    semanas_anteriores = request.query_params.get('semanas_anteriores')
 
     queryset = PautaHistorico.objects
-
     date = None
+
     if data_referencia:
         try:
             date = datetime.strptime(data_referencia, '%Y-%m-%d')
         except ValueError:
             print(
-                f'Data de referência ({data_referencia}) inválida. '
-                'Utilizando data atual como data de referência.')
+                f'Data de referência ({data_referencia}) inválida. ')
+            date = datetime.datetime.now().strptime('%Y-%m-%d')
         else:
-            queryset = queryset.filter(data__lte=date)
-
-    if semanas_anteriores:
-        if not date:
-            date = datetime.today()
-        start_date = date - timedelta(weeks=int(semanas_anteriores))
-        queryset = queryset.filter(data__gte=start_date)
+            queryset = queryset.filter(data__gte=date)
```

Comment:
O `else` só é executado quando o `try` não gera o `except`. 
É isso mesmo que se quer aqui? Se a data de referência não for entendida o `filter(data__gte=date)`, usando `now()`, não será executado.
Estou perguntando porque não sei a resposta mesmo...

Reply:
Modifiquei o código, acho que estava muito confuso. Deixei para quando ele passar uma data errada, a api retornar um erro mesmo.